.github/workflows/ci.yml

```diff
@@ -0,0 +1,245 @@
+name: CI
```

Comment:
I think it might worth considering to split this file into smaller pieces two, that can help to run them in parallel even more and maybe put more _ref_ related restriction on them. May be something like this:

```
$ tree .github/workflows/
.github/workflows/
├── prerelease.yaml   (if needed)
├── publish.yaml      (for building, publishing, promoting)
├── release.yaml      (if needed)
└── validate.yaml     (for lint, vendor check, tests)
```

Reply:
It's currently challenging to break some of these up into distinct workflows (or even jobs) because we're using the Docker daemon to pass state between steps; i.e. one step builds a Docker image, then the next step publishes it, etc. This could be an area for improvement in future though.

Reply:
Yes for sure, this is a really good candidate for future improvements when the initial phase of unblocking the faulty CI has passed. :+1: